ynr/apps/ynr_refactoring/views.py

```diff
@@ -0,0 +1,104 @@
+from django.views.generic import RedirectView
+from django.views.defaults import page_not_found
+from django.core.urlresolvers import reverse
+from django.shortcuts import get_object_or_404
+
+from elections.models import Election
+from candidates.models import PostExtraElection
+from parties.models import Party
+
+from .models import PageNotFoundLog
+
+
+def logged_page_not_found_wrapper(request, *args, **kwargs):
+    # Log stuff
+    PageNotFoundLog(url=request.path).save()
+    return page_not_found(request, *args, **kwargs)
+
+
+# Redirect views
+
+
+class PermanentRedirectView(RedirectView):
+    permanent = True
+
+
+class RedirectConstituencyListView(PermanentRedirectView):
+    """
+    Moves from `/election/<election>/constituencies` URLs to
+    /elections/<election>/ URLs
+
+    """
+
+    def get_redirect_url(self, *args, **kwargs):
+
+        election = get_object_or_404(Election, slug=self.kwargs["election"])
+        url = election.get_absolute_url()
+
+        args = self.request.META.get("QUERY_STRING", "")
+        if args and self.query_string:
+            url = "%s?%s" % (url, args)
+        return url
+
+
+class RedirectPostsListView(PermanentRedirectView):
+    """
+    Moves the /posts URL to /elections/
+
+    """
+
+    url = "/elections/"
+
+
+class RedirectConstituenciesUnlockedView(PermanentRedirectView):
+    """
+    Move /election/<election>/constituencies/unlocked to
+    /elections/<election>/unlocked/
+
+    """
+
+    def get_redirect_url(self, *args, **kwargs):
+        return reverse(
+            "constituencies-unlocked",
+            kwargs={"election": self.kwargs["election"]},
+        )
+
+
+class RedirectConstituencyDetailView(PermanentRedirectView):
+    def get_redirect_url(self, *args, **kwargs):
+
+        ballot = get_object_or_404(
+            PostExtraElection,
+            election__slug=self.kwargs["election"],
+            post__slug=self.kwargs["post_id"],
+        )
+        url = ballot.get_absolute_url()
+
+        args = self.request.META.get("QUERY_STRING", "")
+        if args and self.query_string:
+            url = "%s?%s" % (url, args)
+        return url
+
+
+class RedirectConstituencyDetailCSVView(PermanentRedirectView):
```

Comment:
Not sure what is going on here, but this one is't working as expected. If I call https://candidates.democracyclub.org.uk/election/local.fareham.2017-11-09/post/DIW:E05004527/stubbington.csv that spits out a CSV, but if I check this out locally and call http://127.0.0.1:8000/election/local.fareham.2017-11-09/post/DIW:E05004527/stubbington.csv its dumping me to a non-csv view. Not looked into it in more detail than that, but maybe the ".csv" bit is being greedy captured as part of the ignored slug or something?

Reply:
Good catch, thanks. I've just pushed 742ec5b14 to fix this.